sdk/python/pvsite_datamodel/read/generation.py

```diff
@@ -68,19 +72,24 @@ def get_pv_generation_by_sites(
     # start main query
     query = session.query(GenerationSQL)
     query = query.join(SiteSQL)
-    query = query.join(DatetimeIntervalSQL)
 
     # Filter by time interval
-    query = filter_query_by_datetime_interval(query=query, start_utc=start_utc, end_utc=end_utc)
+    if start_utc is not None:
```

Comment:
could refactor this, but I actually quite like the readabiliy of having this all here